signer/src/config.rs

```diff
@@ -50,16 +152,82 @@ pub struct BlockNotifierConfig {
     pub subscribe_interval: u64,
 }
 
+impl Validatable for BlockNotifierConfig {
+    fn validate(&self) -> Result<(), ConfigError> {
+        if self.server.is_empty() {
+            return Err(ConfigError::Message(
+                "[block_notifier] Electrum server cannot be empty".to_string(),
+            ));
+        }
+
+        Ok(())
+    }
+}
+
+/// Signer-specific configuration
+#[derive(Deserialize, Clone, Debug)]
+pub struct SignerConfig {
+    /// Stacks account configuration. This is the account that the signer will
+    /// use to identify itself on the network and sign transactions.
+    pub stacks_account: StacksAccountConfig,
+    /// P2P network configuration
+    pub p2p: P2PNetworkConfig,
+}
+
+impl Validatable for SignerConfig {
+    fn validate(&self) -> Result<(), ConfigError> {
+        self.p2p.validate()?;
+        Ok(())
+    }
+}
+
+/// Keypair configuration
+#[derive(Deserialize, Clone, Debug)]
+pub struct StacksAccountConfig {
+    /// The private key of the signer
+    #[serde(deserialize_with = "private_key_deserializer")]
+    pub private_key: PrivateKey,
```

Comment:
I think we can just derive `Deserialize` and use `#[serde(transparent)]` now that we've settled on only supporting 32 byte private keys, no need for the `deserialize_with`.

Reply:
Disagree in this case as this is the only way to return more informative errors to the end-user if this field is of type `PrivateKey` and not `String` (so we can do our own validation pre-deserialize). When an end-user is involved, more information is almost always better as it can reduce potential support. And in this particular case I think it's very likely that there will be users trying to paste a 33-byte key and the `SecretKey`'s "malformed key" message is less than informative.

Reply:
👍🏽 Yeah that makes perfect sense. I should have updated this comment after https://github.com/stacks-network/sbtc/pull/428#discussion_r1720380323 to state that I think it's better to tell the user what the issue was.